src/components/Classes/components/JoinClassModal.js

```diff
@@ -1,89 +1,82 @@
-import React from 'react';
+import React, { useState } from 'react';
 import ReactModal from 'react-modal';
 import { Redirect } from 'react-router-dom';
 import {
   Button, Container, Row, Col, FormGroup, Label, Input,
 } from 'reactstrap';
-import * as fetch from '../../../lib/fetch.js';
+import * as fetch from '../../../lib/fetch';
 
-class JoinClassModal extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      wid: '',
-      disableSubmit: false,
-      error: '',
-      redirect: false,
-    };
-  }
+function JoinClassModal({ props }) {
+  const [wid, setWid] = useState('');
+  const [_disableSubmit, setDisableSubmit] = useState(false);
+  const [error, setError] = useState('');
+  const [redirect, setRedirect] = useState(false);
 
-  closeModal = () => {
-    if (this.props.onClose && {}.toString.call(this.props.onClose) === '[object Function]') {
-      this.props.onClose();
+  const closeModal = () => {
+    const { onClose } = props;
```

Comment:
would you like to destructure all props outside of this method? so something like on line 14
```
const {onClose, uid, ...} = props
```